source/common/upstream/cluster_manager_impl.cc

```diff
@@ -179,13 +179,15 @@ void ClusterManagerImpl::initializeClustersFromV1Json(const Json::Object& config
 }
 
 void ClusterManagerImpl::initializeClustersFromV2Proto(const envoy::api::v2::Bootstrap& bootstrap) {
-  for (const auto& cluster : bootstrap.bootstrap_clusters()) {
-    loadCluster(cluster, false);
+  if (bootstrap.has_static_resources()) {
```

Comment:
nit: I think @htuch has only been doing `has*` when it's really necessary. Here I think you can just call static_resources() to construct a default, the clusters() will just return empty list.